lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -459,6 +466,8 @@ RecursiveTypeProperties FIRRTLBaseType::getRecursiveTypeProperties() const {
       .Case<FEnumType>([](FEnumType enumType) {
         return enumType.getRecursiveTypeProperties();
       })
+      .Case<RefType>(
```

Comment:
Might be nice to combine `BundleType`, `FVectorType`, `FEnumType`, and `RefType` into a single `Case`

Reply:
Totally, thanks.  I think it was this way to ensure each had concrete type, but hopefully can use `auto` (seemed to work in follow-on, but didn't check, say, MSVC).  Let's find out!